components/src/main/java/uk/gov/hmrc/components/organism/editable/EditableListView.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.hmrc.components.organism.editable
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import androidx.constraintlayout.widget.ConstraintLayout
+import uk.gov.hmrc.components.R
+import uk.gov.hmrc.components.databinding.ComponentEditableListViewBinding
+import uk.gov.hmrc.components.extensions.setAccessibilityMessage
+
+open class EditableListView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+) : ConstraintLayout(context, attrs) {
+
+    private val binding: ComponentEditableListViewBinding =
+        ComponentEditableListViewBinding.inflate(LayoutInflater.from(context), this, true)
+
+    data class EditableItem(var name: String, var value: String)
+
+    lateinit var editableListViewAdapter: EditableListViewAdapter
+    var editableItem = ArrayList<EditableItem>()
+    var itemPosition: Int? = null
+    var clicked = false
+
+    init {
+
+        binding.secondaryButton.apply {
+
+            setCompoundDrawablesWithIntrinsicBounds(R.drawable.ic_edit, 0, 0, 0)
```

Comment:
If we use the `IconButton` component rather than `SecondaryButton` then we don't need to add the icon in this way. However, I think it does still need `setIconResource(icon)` (see `IconButtonCardView` for an example of this)